### PR TITLE
#2: Charge player for buys

### PR DESCRIPTION
**Pull request by bbqbaron**
First PR!

For food cards:
* charge player for buys
* no-op on buys that can't happen
* show X when you can't afford a card
* tests with Jest, which is popular enough that I hope it's OK

* I super-presumptuously added [Prettier](https://prettier.io/) which all JS hipsters are mad about. Happy to pull it out! I've found it's the best thing ever. TLDR it autoformats your code so you can write like a drunk monkey and still get consistent style. To try it out, run `yarn prettier -- --write app/**/*.js*`; we can make it a package script if you like it.

### CODE REVIEW

File: app/reducers/gameState.test.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { buyFood } from '../actions';
```
Comment (bbqbaron) [inline app/reducers/gameState.test.js]: Tests!

File: app/reducers/gameState.js [hunk 2]
```diff
@@ -7,59 +7,79 @@ const baseState = {
...
+  } = state;
   switch (action.type) {
-    case "APPLY_MANA":
-      return Object.assign({}, state, {mana: state.mana + action.num});
-    case "BUY_FOOD":
+    case 'APPLY_MANA':
+      return Object.assign({}, state, { mana: state.mana + action.num });
+    case 'BUY_FOOD':
       card = state.foodDeck.find(c => c.id == action.id);
+
```
Comment (bbqbaron) [inline app/reducers/gameState.js]: Reducer change.

File: app/components/FoodDeck.jsx [hunk 3]
```diff
@@ -38,32 +38,33 @@ class FoodDeck extends Component {
...
+    const { foodDeck } = this.props.gameState;
     const FACEUP_COUNT = 4;
     const faceup = foodDeck.slice(-FACEUP_COUNT);
     let remaining = foodDeck.length - FACEUP_COUNT;
     if (remaining < 0) remaining = 0;
 
-    const faceupList = faceup.map(c => <li key={c.id} className="faceup-item"><FoodCard dragType="buyCard" {...c} /></li>);
+    const faceupList = faceup.map(c => (
+      <li key={c.id} className="faceup-item">
+        <FoodCard dragType="buyCard" mana={this.props.gameState.mana} {...c} />
```
Comment (bbqbaron) [inline app/components/FoodDeck.jsx]: Passing state mana to cards so they know to be buyable or not.

File: app/components/FoodCard.jsx [hunk 4]
```diff
@@ -1,50 +1,53 @@
...
-      mounted: false
+      mounted: false,
     };
   }
 
   render() {
+    const { isDragging, connectDragSource, mana, cost } = this.props;
 
-    const {isDragging, connectDragSource} = this.props;
+    const canAfford = mana >= cost;
```
Comment (bbqbaron) [inline app/components/FoodCard.jsx]: Show Xes or not, with accompanying minor style update.

File: app/components/FoodDeck.jsx (hunk 3 above)
Comment (jhmullen) [reply app/components/FoodDeck.jsx]: A quick question on this, more for my own edification.  Is there any reason not to reach directly into gameState.mana from FoodCard.jsx?  This prevents us from having to "prime" all FoodCard instances with what is already a globally known value.

**Comment by jhmullen** [comment]
Re: Library choices

Truth be told, I have little experience with writing tests (boo hiss I know) so I'm glad to follow your lead on Jest.  

For my main react project at work we are using a jslinter, I'm definitely not opposed to a style enforcer, especially an automagical one like Prettier.

File: app/components/FoodDeck.jsx (hunk 3 above)
Comment (bbqbaron) [reply app/components/FoodDeck.jsx]: I think this is an evolving idiom in the Redux community: how many components are "aware" and how many are not? Do we connect to the store everywhere such that Redux is now inextricable from the app or do we pass a lot of props around? etc etc. Basically, judgment call 😄 .

**Pull request merged**